File: js/src/maplestory/api/dto/character/characterSkillInfoDto.ts
import { CharacterSkillInfoDtoBody } from '../../response/character/characterSkillDtoBody';

/**
 * 캐릭터 스킬 상세 정보
 */
class CharacterSkillInfoDto {
  /**
   * 스킬 명
   */
  skillName: string;

  /**
   * 스킬 설명
   */
  skillDescription: string;

  /**
   * 스킬 레벨
   */
  skillLevel: number;

  /**
   * 스킬 레벨 별 효과 설명
   */
  skillEffect: string;

  /**
   * 다음 스킬 레벨 효과 설명
   */
  skillEffectNext: string | null;

  /**
   * 스킬 아이콘
   */
  skillIcon: string;

  constructor(obj: CharacterSkillInfoDtoBody) {
    const {
      skill_name,
      skill_description,
      skill_level,
      skill_effect,
      skill_effect_next,
      skill_icon,
    } = obj;

    this.skillName = skill_name;
    this.skillDescription = skill_description;
    this.skillLevel = skill_level;
    this.skillEffect = skill_effect;
    this.skillEffectNext = skill_effect_next;
    this.skillIcon = skill_icon;
  }
}

export { CharacterSkillInfoDto };

File: java/src/main/java/dev/spiralmoon/maplestory/api/msea/dto/character/CharacterSkillInfoDTO.java
package dev.spiralmoon.maplestory.api.msea.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

/**
 * Skill information
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterSkillInfoDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterSkillInfoDTO {

    /**
     * Skill name
     */
    @JsonProperty("skill_name")
    private String skillName;

    /**
     * Skill description
     */
    @JsonProperty("skill_description")
    private String skillDescription;

    /**
     * Skill level
     */
    @JsonProperty("skill_level")
    private Long skillLevel;

    /**
     * Effect description by skill level
     */
    @JsonProperty("skill_effect")
    private String skillEffect;

    /**
     * Effect description for the next skill level
     */
    @JsonProperty("skill_effect_next")
    private String skillEffectNext;

    /**
     * Skill icon
     */
    @JsonProperty("skill_icon")
    private String skillIcon;
}
